### PR TITLE
[#69] [iOS] [UI] As a user, I can see dropdown answer

### LINKED ISSUE #45: [iOS] [UI] As a user, I can see dropdown answer
## Why

Dropdown questions are one of the most common question types to allow users to select one answer. The dropdown options could be anything, such as, a set of cities, locations, or even yes/no, since they are text-based options.

Even though this is called a dropdown, to provide the mobile experience, we will use a native picker wheel for this type of question.

## Acceptance Criteria

For dropdown questions,
- Display the answers on a native picker wheel
- Highlight the selected one
- Gray out the non-selected options

## Design

[Dropdown Question](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=48%3A0&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by suho**
- Closes #45 

## What happened 👀

Add a dropdown answer view for the survey.

## Insight 📝

For the iOS side, I used the built-in component to implement the dropdown view.

Noted: I created Answer and AnswerInput models because we need for view (SwiftUI). Without it, we can write code, but most of the time, there are no auto-suggestions or previews with SwiftUI 🙈

### CODE REVIEW

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/DropdownAnswerView.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                let font = selectedInput == id
+                    ? Typography.neuzeitSLTStdBookHeavy.font(size: 20.0)
+                    : Typography.neuzeitSLTStdBook.font(size: 20.0)
+                Text(answers.first(where: { $0.id == id })?.content ?? "")
+                    .font(font)
+                    .foregroundColor(Color.white)
+            }
+        }
+        .pickerStyle(.wheel)
+        .padding(20.0)
```
Comment (vnntsu) [inline ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/DropdownAnswerView.swift:34]: Just a concern relating to the space between the 2 items. Could we increase it to match the design?

**Review by vnntsu — APPROVED** [review]
Apart from a concern. All good 🚀

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/DropdownAnswerView.swift (hunk 1 above)
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/DropdownAnswerView.swift:34]: We can't. We are using the default picker, it's not supported for fully customization 😢

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Models/AnswerInput.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    case select(id: String)
+    case content(id: String, value: String)
+
+    var id: String {
+        switch self {
+        case let .select(id):
+            return id
+        case let .content(id, _):
+            return id
```
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Models/AnswerInput.swift]: Fixed in https://github.com/suho/kmm-ic/pull/130/commits/9d5e96ffde69e35ba27cbd970e26d2ae9b1de415

**Review by nkhanh44 — COMMENTED** [review]
The dropdown answer doesn't look like the AC 💭 . Please update the UI to the same as the design 🙏 .
Screenshot 2023-01-10 at 13 29 12

**Comment by suho** [comment]
@nkhanh44 On [the AC](https://github.com/suho/kmm-ic/issues/45), I mentioned preferring using the native picker wheel. So please ignore this :D

**Review by nkhanh44 — APPROVED** [review]
go go 🚀

**Pull request merged**